ui/store/actions.ts

```diff
@@ -2427,6 +2427,19 @@ export function hideIpfsModal(): Action {
     type: actionConstants.SHOW_IPFS_MODAL_CLOSE,
   };
 }
+
+export function showSitePermissionsModal(): Action {
+  return {
+    type: actionConstants.SHOW_SITE_PERMISSIONS_MODAL,
+  };
+}
+
+export function hideSitePermissionsModal(): Action {
+  return {
+    type: actionConstants.HIDE_SITE_PERMISSIONS_MODAL,
+  };
+}
+
```

Comment:
Do you think we can move this to local state in modal?

Reply:
Echoing Nidhi's question here -- can showing and hiding this functionality be done by local state (aka `useState`) of a parent component, or can this modal be launched from multiple places in the app?

Reply:
Added [5f4be6f](https://github.com/MetaMask/metamask-extension/pull/20633/commits/5f4be6fa1abef51907b06bb264da7f25bb2642bc)